packages/components/src/internal/components/navigation/FolderMenu.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { FC, memo, useCallback, useEffect, useState } from 'react';
+import { ActionURL } from '@labkey/api';
+import { Dropdown, MenuItem } from 'react-bootstrap';
+
+import {
+    Alert,
+    buildURL,
+    isLoading,
+    LoadingSpinner,
+    LoadingState,
+    resolveErrorMessage,
+    useServerContext,
+} from '../../..';
+
+import { ComponentsAPIWrapper, getDefaultAPIWrapper } from '../../APIWrapper';
+import { getCurrentAppProperties } from '../../app/utils';
+import { AppProperties } from '../../app/models';
+import { blurActiveElement } from '../../util/utils';
+
+interface FolderMenuItem {
+    href: string;
+    id: string;
+    label: string;
+}
+
+interface Props {
+    api?: ComponentsAPIWrapper;
+    appProperties?: AppProperties;
+}
+
+export const FolderMenu: FC<Props> = memo(({ api, appProperties }) => {
+    const { controllerName } = appProperties;
+    const [error, setError] = useState<string>();
+    const [loading, setLoading] = useState<LoadingState>(LoadingState.INITIALIZED);
+    const [open, setOpen] = useState(false);
+    const [items, setItems] = useState<FolderMenuItem[]>([]);
+    const hasError = !!error;
+    const isLoaded = !isLoading(loading);
+    const { container } = useServerContext();
+
+    useEffect(() => {
+        setLoading(LoadingState.LOADING);
+        setError(undefined);
+
+        (async () => {
+            try {
+                const folders = await api.security.fetchContainers({
+                    containerPath: container.type === 'project' ? container.path : container.parentPath,
+                });
+
+                const items_: FolderMenuItem[] = folders.map(folder => ({
+                    href: buildURL(controllerName, `${ActionURL.getAction()}.view`, undefined, {
+                        container: folder.path,
+                        returnUrl: false,
+                    }),
+                    id: folder.id,
+                    label: folder.name,
```

Comment:
any reason to use the `folder.title` instead of the `folder.name` here? I don't think many users are setting the title to be something different then the name.

Reply:
Switched to use title.